lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -1162,7 +1162,7 @@ def _create_node_arm_parameters(
         arm_parameters = AzureNodeArmParameter.from_node_runbook(runbook)
 
         os_disk_size = 30
-        if arm_parameters.vhd:
+        if arm_parameters.vhd and len(arm_parameters.vhd) > 0:
```

Comment:
It doesn't need to check the length of a string. `if str_value` works like False for both None and "".

Reply:
I was seeing really weird behavior during testing where arm_parameters.vhd == "" and it would still evaluate to true. I'm not sure if it had something to do with arm_parameters.vhd being a `@property` where maybe it evaluated as true while it was still a function handle.

Reply:
I had a quick test, the property works like expected.

```python
class A:
    @property
    def b(self) -> str:
        return "aaa"


x = A()
print(x.b)
if x.b:
    print("if ")
else:
    print("else")

```

Reply:
Resolved